src/textUtils.c

```diff
@@ -101,7 +116,65 @@ size_t str_formatTtl(uint64_t ttl, char* out, size_t outSize)
 	return strlen(out);
 }
 
+// returns length of the resulting string
 size_t str_formatMetadata(const uint8_t* metadataHash, size_t metadataHashSize, char* out, size_t outSize)
 {
 	return encode_hex(metadataHash, metadataHashSize, out, outSize);
 }
+
+// we only check if it is non-zero ASCII
+void str_validateText(const uint8_t* url, size_t urlSize)
+{
+	ASSERT(urlSize < BUFFER_SIZE_PARANOIA);
+
+	for (size_t i = 0; i < urlSize; i++) {
+		VALIDATE(url[i] <= 127, ERR_INVALID_DATA);
+		VALIDATE(url[i] > 0, ERR_INVALID_DATA);
+	}
+}
+
+#ifdef DEVEL
+// only used in internal device tests
+size_t urlToBuffer(const char* url, uint8_t* buffer, size_t bufferSize)
+{
+	size_t urlLength = strlen(url);
+	ASSERT(urlLength < BUFFER_SIZE_PARANOIA);
+	ASSERT(bufferSize < BUFFER_SIZE_PARANOIA);
+	ASSERT(bufferSize >= urlLength);
+
+	const char* validChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._~:/?#[]@!$&'()*+,;%=";
```

Comment:
is there any point in validating the input for a function used exclusively for tests?

Reply:
That's a good question. How do we do validation in real code? Also, looking below, maybe we should have one
`validateStringContainsOnlyAllowedChars(string, char_string)` function

Reply:
This is kind of a leftover from earlier times. I rewrote it in 25aa9200a2c2b52fea48c4f9b51aa497be332a97.